lib/base_service.ts

```diff
@@ -132,9 +134,26 @@ export class BaseService {
     if (this._options.url) {
       _credentials.url = this._options.url;
     }
+    if (this._options.access_token) {
+      _credentials.access_token = this._options.access_token;
+    }
     return _credentials;
   }
 
+  /**
+   * Set an IAM access token to authenticate the service with
+   * The SDK will assume this token is valid and that this method
+   * will be used to set a new token upon expiration.
+   *
+   * @param {string} accessToken - Valid IAM access token
+   * @returns {void}
+   */
+  public setAccessToken(accessToken: string) {
```

Comment:
Should this be `accessToken` or `access_token`? I don't have a preference, but want to make sure the API is consistent.

Reply:
`tslint` yells at you if an argument name is not camel case, which is why I used `accessToken`. Also, while it is public facing, it takes in a pure string and the user does not have to identify it with `accessToken`. The constructor takes in an object and requires the identifier `access_token`. So while the argument names are not consistent, I would say the API technically is.

That being said, the difference would be reflected in the documentation and I am happy to change the arg name and add a `tslint-disable-line` comment if that would be preferred.

Reply:
I agree with @glennrfisher. You can disable the rule for that line.

Reply:
The style rules for Javascript say that we should be using [camel case for naming objects, functions, and instances](https://github.com/airbnb/javascript#naming--camelCase).  Snake case violates the style guide.

That said, I think we have not been doing a good job of generating SDKs that follow this style, so I'm not sure this is worth arguing over.